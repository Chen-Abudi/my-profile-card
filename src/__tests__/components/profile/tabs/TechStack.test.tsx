import React from "react";
import { render, RenderResult } from "@testing-library/react";
import "@testing-library/jest-dom";

import TechStackTab from "@/components/profile/tabs/TechStack";

describe("TechStackTab", () => {
  let component: RenderResult;

  beforeEach(() => {
    component = render(<TechStackTab />);
  });

  it("should render Tech Stack section", () => {
    const { getByText } = component;

    expect(getByText("Tech Stack")).toBeInTheDocument();
  });

  it("should render Programming Languages section", () => {
    const { getByText } = component;

    expect(getByText("Programming Languages:")).toBeInTheDocument();
  });

  it("should render all Programming Languages items", () => {
    const { getByText } = component;

    ["TypeScript", "JavaScript", "Java", "HTML", "CSS", "SCSS", "Sass"].forEach(
      (item) => {
        expect(getByText(item)).toBeInTheDocument();
      }
    );
  });

  it("should render Frontend Frameworks & Libraries section", () => {
    const { getByText } = component;

    ["React", "Vue.js", "Next.js", "Vite", "TailwindCSS", "Redux"].forEach(
      (item) => {
        expect(getByText(item)).toBeInTheDocument();
      }
    );
  });

  it("should render Backend Frameworks section", () => {
    const { getByText } = component;

    expect(getByText("Backend Frameworks:")).toBeInTheDocument();
  });

  it("should render all Backend Frameworks items", () => {
    const { getByText } = component;

    ["Node.js", "Express.js", "Spring", "Spring Boot"].forEach((item) => {
      expect(getByText(item)).toBeInTheDocument();
    });
  });

  it("should render Databases section", () => {
    const { getByText } = component;

    expect(getByText("Databases:")).toBeInTheDocument();
  });

  it("should render all Databases items", () => {
    const { getByText } = component;

    ["SQL & NoSQL", "PostgreSQL", "MongoDB", "Firebase", "SQLite"].forEach(
      (item) => {
        expect(getByText(item)).toBeInTheDocument();
      }
    );
  });

  it("should render DevOps & CI/CD section", () => {
    const { getByText } = component;

    expect(getByText("DevOps & CI/CD:")).toBeInTheDocument();
  });

  it("should render all DevOps & CI/CD items", () => {
    const { getByText } = component;

    expect(getByText("Docker")).toBeInTheDocument();
  });

  it("should render all Version Control section", () => {
    const { getByText } = component;

    expect(getByText("Version Control:")).toBeInTheDocument();
  });

  it("should render all Version Control items", () => {
    const { getByText } = component;

    ["Git", "GitHub", "GitLab"].forEach((item) => {
      expect(getByText(item)).toBeInTheDocument();
    });
  });

  it("should render Design Tools section", () => {
    const { getByText } = component;

    expect(getByText("Design Tools:")).toBeInTheDocument();
  });

  it("should render all Design Tools items", () => {
    const { getByText } = component;

    ["Figma", "Canva"].forEach((item) => {
      expect(getByText(item)).toBeInTheDocument();
    });
  });

  it("should render Testing Tools section", () => {
    const { getByText } = component;

    expect(getByText("Testing Tools:")).toBeInTheDocument();
  });

  it("should render all Testing Tools items", () => {
    const { getByText } = component;

    ["Jest", "React Testing Library", "JUnit"].forEach((item) => {
      expect(getByText(item)).toBeInTheDocument();
    });
  });

  it("should match snapshot", () => {
    const { container } = component;

    expect(container).toMatchSnapshot();
  });
});
