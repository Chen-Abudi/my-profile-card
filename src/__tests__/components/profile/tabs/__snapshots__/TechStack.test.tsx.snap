// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TechStackTab should match snapshot 1`] = `
<div>
  <div>
    <h2
      class="text-xl font-bold"
    >
      Tech Stack
    </h2>
    <ul
      class="list-disc pl-5"
    >
      <li>
        <strong>
          Programming Languages:
        </strong>
        <ul
          class="list-disc pl-5"
        >
          <li>
            TypeScript
          </li>
          <li>
            JavaScript
          </li>
          <li>
            Java
          </li>
          <li>
            HTML
          </li>
          <li>
            CSS
          </li>
          <li>
            SCSS
          </li>
          <li>
            Sass
          </li>
        </ul>
      </li>
      <li>
        <strong>
          Frontend Frameworks & Libraries:
        </strong>
        <ul
          class="list-disc pl-5"
        >
          <li>
            React
          </li>
          <li>
            Vue.js
          </li>
          <li>
            Next.js
          </li>
          <li>
            Vite
          </li>
          <li>
            TailwindCSS
          </li>
          <li>
            Redux
          </li>
        </ul>
      </li>
      <li>
        <strong>
          Backend Frameworks:
        </strong>
        <ul
          class="list-disc pl-5"
        >
          <li>
            Node.js
          </li>
          <li>
            Express.js
          </li>
          <li>
            Spring
          </li>
          <li>
            Spring Boot
          </li>
        </ul>
      </li>
      <li>
        <strong>
          Databases:
        </strong>
        <ul
          class="list-disc pl-5"
        >
          <li>
            SQL & NoSQL
          </li>
          <li>
            PostgreSQL
          </li>
          <li>
            MongoDB
          </li>
          <li>
            Firebase
          </li>
          <li>
            SQLite
          </li>
        </ul>
      </li>
      <li>
        <strong>
          DevOps & CI/CD:
        </strong>
        <ul
          class="list-disc pl-5"
        >
          <li>
            Docker
          </li>
        </ul>
      </li>
      <li>
        <strong>
          Version Control:
        </strong>
        <ul
          class="list-disc pl-5"
        >
          <li>
            Git
          </li>
          <li>
            GitHub
          </li>
          <li>
            GitLab
          </li>
        </ul>
      </li>
      <li>
        <strong>
          Design Tools:
        </strong>
        <ul
          class="list-disc pl-5"
        >
          <li>
            Figma
          </li>
          <li>
            Canva
          </li>
        </ul>
      </li>
      <li>
        <strong>
          Testing Tools:
        </strong>
        <ul
          class="list-disc pl-5"
        >
          <li>
            Jest
          </li>
          <li>
            React Testing Library
          </li>
          <li>
            JUnit
          </li>
        </ul>
      </li>
    </ul>
  </div>
</div>
`;
